src/schema/events.ts

```diff
@@ -398,63 +399,70 @@ builder.mutationFields((t) => ({
-        address,
-        latitude,
-        longitude,
-        meetingURL,
-        status,
-        timeZone,
-      } = input;
-      if (!ctx.USER) throw new Error("User not found");
-      if (await canCreateEvent(ctx.USER.id, communityId, ctx.DB))
-        throw new Error("You don't have permission to create an event");
-      const result = await ctx.DB.transaction(async (trx) => {
-        try {
-          const id = v4();
-          const newEvent = insertEventsSchema.parse({
-            id,
-            name,
-            description,
-            visibility,
-            startDateTime,
-            endDateTime,
-            maxAttendees,
-            geoAddressJSON: address,
-            geoLongitude: longitude,
-            getLatitude: latitude,
-            meetingURL,
-            status: status ?? "inactive",
-            timeZone,
-          });
+      try {
+        const {
+          name,
+          description,
+          visibility,
+          startDateTime,
+          endDateTime,
+          communityId,
+          maxAttendees,
+          address,
+          latitude,
+          longitude,
+          meetingURL,
+          status,
+          timeZone,
+        } = input;
+        if (!ctx.USER) throw new Error("User not found");
+        if (!(await canCreateEvent(ctx.USER.id, communityId, ctx.DB))) {
+          throw new Error("FORBIDDEN");
+        }
+        const result = await ctx.DB.transaction(async (trx) => {
+          try {
+            const id = v4();
+            const newEvent = insertEventsSchema.parse({
+              id,
+              name,
+              description,
+              visibility,
+              startDateTime,
+              endDateTime,
+              maxAttendees,
+              geoAddressJSON: address,
+              geoLongitude: longitude,
+              getLatitude: latitude,
+              meetingURL,
+              status: status ?? "inactive",
+              timeZone,
+            });
 
-          const events = await trx
-            .insert(eventsSchema)
-            .values(newEvent)
-            .returning()
-            .get();
-          await trx
-            .insert(eventsToCommunitiesSchema)
-            .values({
-              eventId: id,
-              communityId: communityId,
-            })
-            .returning()
-            .get();
+            const events = await trx
+              .insert(eventsSchema)
+              .values(newEvent)
+              .returning()
+              .get();
+            await trx
+              .insert(eventsToCommunitiesSchema)
+              .values({
+                eventId: id,
+                communityId: communityId,
+              })
+              .returning()
+              .get();
 
-          return events;
-        } catch (e) {
-          // trx.rollback();
-        }
-      });
-      return selectEventsSchema.parse(result);
+            return events;
+          } catch (e) {
+            // trx.rollback();
```

Comment:
creo q deberiamos hacer un rollback aca, onda si algo falló que no pase nada en la BDD.

Reply:
Por otro lado...es un puro insert... mejor removamos el rollbacK 😅 

en esta caso, si el insert falla. lo va a capturar el try catch de la linea `423` ...pero despues to tira otro error, entonces el try/catch q pusiste no va a capturar nada :/

Reply:
Agregado, hice otro graphql error en esa seccion